core/src/main/java/org/geysermc/geyser/translator/protocol/bedrock/entity/player/input/BedrockMovePlayer.java

```diff
@@ -219,6 +226,47 @@ static void translate(GeyserSession session, PlayerAuthInputPacket packet) {
         if (entity.getRightParrot() != null) {
             entity.getRightParrot().moveAbsolute(entity.getPosition(), entity.getYaw(), entity.getPitch(), entity.getHeadYaw(), true, false);
         }
+
+        // If player is inside a vehicle or the player itself CAN'T be push then don't push them.
+        if (entity.getVehicle() == null && entity.isPushable(session)) {
+            // TODO: Is looping through every entities a good idea?
+            for (Entity other : session.getEntityCache().getEntities().values()) {
+                if (other == entity)  {
+                    return;
+                }
+
+                final BoundingBox entityBoundingBox = new BoundingBox(0, 0, 0, other.getBoundingBoxWidth(), other.getBoundingBoxHeight(), other.getBoundingBoxWidth());
+                entityBoundingBox.translate(other.position().toDouble());
+
+                // If this entity can't collide with the player or isn't near the player or is the player itself, continue.
+                if (!entityBoundingBox.checkIntersection(session.getCollisionManager().getPlayerBoundingBox()) || !other.isPushable(session)) {
+                    continue;
+                }
+
+                // According to Entity#push line 1563 (Mojmap).
```

Comment:
these will change between updates - i'd remove the line and instead note the minecraft java version this is from